spec/time_spec.rb

```diff
@@ -95,6 +95,87 @@ module ICU
           expect(f4.date_format(true)).to eq("MMM y")
         end
       end
+
+      context 'hour cycle' do
+        # en_AU normally is 12 hours, fr_FR is normally 23 hours
+        ['en_AU', 'fr_FR', 'zh_CN'].each do |locale_name|
+          context "with locale #{locale_name}" do
+            it 'works with hour_cycle: h11' do
+              t = Time.new(2021, 04, 01, 12, 05, 0, "+00:00")
+              str = TimeFormatting.format(t, time: :short, date: :none, locale: locale_name, zone: 'UTC', hour_cycle: 'h11')
+              expect(str).to match(/0:05/i)
+              expect(str).to match(/(pm|下午)/i)
+            end
+            it 'works with hour_cycle: h12' do
+              t = Time.new(2021, 04, 01, 12, 05, 0, "+00:00")
+              str = TimeFormatting.format(t, time: :short, date: :none, locale: locale_name, zone: 'UTC', hour_cycle: 'h12')
+              expect(str).to match(/12:05/i)
+              expect(str).to match(/(pm|下午)/i)
+            end
+            it 'works with hour_cycle: h23' do
+              t = Time.new(2021, 04, 01, 00, 05, 0, "+00:00")
+              str = TimeFormatting.format(t, time: :short, date: :none, locale: locale_name, zone: 'UTC', hour_cycle: 'h23')
+              expect(str).to match(/0:05/i)
+              expect(str).to_not match(/(am|pm)/i)
+            end
+            it 'works with hour_cycle: h24' do
+              t = Time.new(2021, 04, 01, 00, 05, 0, "+00:00")
+              str = TimeFormatting.format(t, time: :short, date: :none, locale: locale_name, zone: 'UTC', hour_cycle: 'h24')
+              expect(str).to match(/24:05/i)
+              expect(str).to_not match(/(am|pm)/i)
+            end
+
+            context '@hours keyword' do
+              before(:each) do
+                skip("Only works on ICU >= 67") if Lib.version.to_a[0] < 67
```

Comment:
Seems to be working for me. Hmm 🤔 

```
2.6.6 :001 > ICU::TimeFormatting.format(Time.now, time: :short, date: :long, locale: 'en-US@hours=h23', hour_cycle: :locale)
 => "April 4, 2022, 22:41 PM"

2.6.6 :002 > ICU::Lib.version.to_s
 => "66.1"
```

Reply:
Yeah it says 22:41 PM which is a bit silly 🙃

Reply:
Oh yes, good catch!